File: 프로그래머스/unrated/181916. 주사위 게임 3/주사위 게임 3.cpp
#include <string>
#include <vector>
#include <math.h>
#include <algorithm>
using namespace std;

int solution(int a, int b, int c, int d) {
    vector<int> nogada;
    nogada.push_back(a);
    nogada.push_back(b);
    nogada.push_back(c);
    nogada.push_back(d);
    sort(nogada.begin(), nogada.end());

    
    if (a == b && a == c && a == d && b == c && b == d && c == d)
        return 1111 * a;
    else if (a != b && a != c && a != d && b != c && b != d && c != d) 
        return *min_element(nogada.begin(), nogada.end());
    else if ((nogada[0] == nogada[1] && nogada[2] == nogada[3]) && nogada[1] != nogada[2]) 
        return (nogada[1] + nogada[2]) * abs(nogada[1] - nogada[2]);
    else if (((nogada[0] == nogada[1] && nogada[1] == nogada[2]) || (nogada[1] == nogada[2] && nogada[2] == nogada[3])) && nogada[0] != nogada[3])
        return count(nogada.begin(), nogada.end(), nogada[0]) > count(nogada.begin(), nogada.end(), nogada[3]) ? (10 * nogada[0] + nogada[3]) * (10 * nogada[0] + nogada[3]) : (10 * nogada[3] + nogada[0]) * (10 * nogada[3] + nogada[0]);
    else if ((nogada[0] == nogada[1] && nogada[2] != nogada[3]) || (nogada[1] == nogada[2] && nogada[0] != nogada[4]) || (nogada[2] == nogada[3] && nogada[0] != nogada[1]))
        return nogada[0] == nogada[1] ? nogada[2] * nogada[3] : nogada[1] == nogada[2] ? nogada[0] * nogada[3] : nogada[0] * nogada[1];
}

File: 프로그래머스/0/181916. 주사위 게임 3/주사위 게임 3.java
import java.util.Arrays;
import java.util.Comparator;
import java.util.List;

class Solution {
    public int solution(int a, int b, int c, int d) {
        List<Integer> list = Arrays.asList(a, b, c, d);
        list.sort(Comparator.naturalOrder());

        a = list.get(0);
        b = list.get(1);
        c = list.get(2);
        d = list.get(3);

        if ((a == b) && (b == c) && (c == d)) {
            return a * 1111;
        }

        if ((a == b && b == c)) {
            return (int) Math.pow((10 * a + d), 2);
        } else if ((b == c && c == d)) {
            return (int) Math.pow((10 * d + a), 2);
        }

        if (a == b && c == d) {
            return (a + c) * Math.abs(a - c);
        } else if (a == b) {
            return c * d;
        } else if (b == c) {
            return a * d;
        } else if (c == d) {
            return a * b;
        }

        return a;
    }
}